Sources/TokamakCore/Views/Canvas/TimelineView.swift

```diff
@@ -0,0 +1,196 @@
+  @inlinable static var everyMinute: EveryMinuteTimelineSchedule { .init() }
+}
+
+public extension TimelineSchedule {
+  static func explicit<S>(_ dates: S) -> ExplicitTimelineSchedule<S>
+    where Self == ExplicitTimelineSchedule<S>, S.Element == Date
+  {
+    .init(dates)
+  }
+}
+
+public struct PeriodicTimelineSchedule: TimelineSchedule {
+  private let entries: Entries
+
+  public struct Entries: Sequence, IteratorProtocol {
+    internal var date: Date
+    internal let interval: TimeInterval
+
+    public mutating func next() -> Date? {
+      defer { date.addTimeInterval(interval) }
+      return date
+    }
+
+    public typealias Element = Date
+    public typealias Iterator = Self
+  }
+
+  public init(from startDate: Date, by interval: TimeInterval) {
+    entries = Entries(date: startDate, interval: interval)
+  }
+
+  public func entries(from startDate: Date, mode: TimelineScheduleMode) -> Entries {
+    entries
+  }
+}
+
+public struct EveryMinuteTimelineSchedule: TimelineSchedule {
+  public struct Entries: Sequence, IteratorProtocol {
+    internal var date: Date
+
+    public mutating func next() -> Date? {
+      defer { date.addTimeInterval(60) }
+      return date
+    }
+
+    public typealias Element = Date
+    public typealias Iterator = Self
+  }
+
+  public init() {}
+
+  public func entries(from startDate: Date,
+                      mode: TimelineScheduleMode) -> EveryMinuteTimelineSchedule.Entries
+  {
+    Entries(date: startDate)
+  }
+}
+
+public struct ExplicitTimelineSchedule<Entries>: TimelineSchedule where Entries: Sequence,
+  Entries.Element == Date
+{
+  private let dates: Entries
+
+  public init(_ dates: Entries) {
+    self.dates = dates
+  }
+
+  public func entries(from startDate: Date, mode: TimelineScheduleMode) -> Entries {
+    dates
+  }
+}
+
+public extension TimelineSchedule where Self == AnimationTimelineSchedule {
+  @inlinable static var animation: AnimationTimelineSchedule { .init() }
+  @inlinable
+  static func animation(
+    minimumInterval: Double? = nil,
+    paused: Bool = false
+  ) -> AnimationTimelineSchedule {
+    .init(minimumInterval: minimumInterval, paused: paused)
+  }
+}
+
+public struct AnimationTimelineSchedule: TimelineSchedule {
+  let minimumInterval: Double?
+  let paused: Bool
+
+  public init(minimumInterval: Double? = nil, paused: Bool = false) {
+    self.minimumInterval = minimumInterval
+    self.paused = paused
+  }
+
+  public func entries(from startDate: Date, mode: TimelineScheduleMode) -> Entries {
+    Entries(startDate: startDate, minimumInterval: minimumInterval, paused: paused)
+  }
+
+  public struct Entries: Sequence, IteratorProtocol {
+    internal let startDate: Date
+    internal let minimumInterval: Double?
+    internal let paused: Bool
+
+    public mutating func next() -> Date? {
+      guard !paused else { return nil }
+      guard let minimumInterval = minimumInterval else {
+        return .init()
```

Comment:
According to the spec, "Pass nil to let the system pick an appropriate update interval." Instead, it looks like this is just updating as quickly as possible when `minimumInterval` is nil. This isn't good for performance.

Additionally, `next()` is a mutating function. It's supposed to change the struct to return a different value next time it's called.